test/test_protocol.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2018-present, Facebook, Inc.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+#
+import dns
+import dns.message
+import struct
+import unittest
+
+from unittest.mock import patch
+from dohproxy.server_protocol import DNSClientProtocolTCP
+
+
+class TCPTestCase(unittest.TestCase):
+
+    def setUp(self):
+        """
+        prepare the dns response
+        """
+        self.dnsq = dns.message.make_query(
+            'www.example.com',
+            dns.rdatatype.ANY)
+        self.dnsr = dns.message.make_response(self.dnsq)
+        self.response = self.dnsr.to_wire()
+
+    def r_receive_helper(self, dnsr):
```

Comment:
you dont need this I believe. Use a mock and check if the mock was called and what with